recipes/aws-c-cal/all/conanfile.py

```diff
@@ -98,4 +98,17 @@ def package_info(self):
         elif self.settings.os in ("FreeBSD", "Linux"):
             self.cpp_info.components["aws-c-cal-lib"].system_libs.append("dl")
         if self._needs_openssl:
-            self.cpp_info.components["_dummy_crypto"].requires.append("openssl::crypto")
+            if self.options["openssl"].shared:
+                self.cpp_info.components["aws-c-cal-lib"].requires.append("openssl::crypto")
+            else:
+                self.cpp_info.components["_dummy_crypto"].requires.append("openssl::crypto")
+                # aws-c-cal dynamically search the openssl symbols.
+                # Mark them as undefined so the linker will include them.
+                # This avoid dynamical look-up for a system crypto library.
+                crypto_symbols = [
+                    "HMAC_CTX_init", "HMAC_CTX_cleanup", "HMAC_Update", "HMAC_Final", "HMAC_Init_ex",
+                ]
+                for crypto_symbol in cryto_symbols:
+                    link_flag = f"-Wl,-u{crypto_symbol}"
+                    self.cpp_info.components["aws-c-cal-lib"].exelinkflags.append(link_flag)
+                    self.cpp_info.components["aws-c-cal-lib"].sharedlinkflags.append(link_flag)
```

Comment:
I think this is clearer, and also fixes an error: I forgot a `,` between `-u` and the symbol.
```suggestion
                crypto_link_flags = "-Wl," + ",".join(f"-u,{symbol}" for symbol in crypto_symbol)
                self.cpp_info.components["aws-c-cal-lib"].exelinkflags.append(crypto_link_flags)
                self.cpp_info.components["aws-c-cal-lib"].sharedlinkflags.append(crypto_link_flags)
```